playbooks/adhoc/upgrades/upgrade.yml

```diff
@@ -9,22 +62,58 @@
     - name: Upgrade base package
       yum: pkg={{ openshift.common.service_type }}{{ openshift_version  }} state=latest
 
-- name: Re-Run cluster configuration to apply latest configuration changes
-  include: ../../common/openshift-cluster/config.yml
+- name: Evaluate oo_first_master
+  hosts: localhost
   vars:
-    g_etcd_group: "{{ 'etcd' }}"
     g_masters_group: "{{ 'masters' }}"
-    g_nodes_group: "{{ 'nodes' }}"
-    openshift_cluster_id: "{{ cluster_id | default('default') }}"
-    openshift_deployment_type: "{{ deployment_type }}"
+  tasks:
+    - name: Evaluate oo_first_master
+      add_host:
+        name: "{{ groups[g_masters_group][0] }}"
+        groups: oo_first_master
+        ansible_ssh_user: "{{ g_ssh_user | default(omit) }}"
+        ansible_sudo: "{{ g_sudo | default(omit) }}"
+      when: g_masters_group in groups and (groups[g_masters_group] | length) > 0
```

Comment:
Since this is currently specifically for the byo style inventory, you don't need to use `g_masters_group` here since we know the masters group is 'masters'.  This whole block is probably not necessary either, since anywhere `oo_first_master` is used you should be able to use `masters[0]` instead